pkg/caveats/errors.go

```diff
@@ -7,6 +7,21 @@ import (
 	"github.com/rs/zerolog"
 )
 
+// EvaluationErr is an error in evaluation of a caveat expression.
+type EvaluationErr struct {
+	error
+}
+
+// MarshalZerologObject implements zerolog.LogObjectMarshaler
+func (err EvaluationErr) MarshalZerologObject(e *zerolog.Event) {
+	e.Err(err.error)
+}
+
+// DetailsMetadata returns the metadata for details for this error.
+func (err EvaluationErr) DetailsMetadata() map[string]string {
+	return map[string]string{}
```

Comment:
no need to allocate, just return `nil` for its zero value

```suggestion
	return nil
```

Reply:
I'm intending to add context here at some point, and I'm not 100% sure our metadata code handles `nil`, so I'd rather just allocate here